pkg/chains/formats/intotoite6/pipelinerun/pipelinerun.go

```diff
@@ -182,8 +188,39 @@ func metadata(pro *objects.PipelineRunObject) *slsa.ProvenanceMetadata {
 }
 
 // add any Git specification to materials
-func materials(pro *objects.PipelineRunObject, logger *zap.SugaredLogger) []slsa.ProvenanceMaterial {
+func materials(pro *objects.PipelineRunObject, logger *zap.SugaredLogger) ([]slsa.ProvenanceMaterial, error) {
 	var mats []slsa.ProvenanceMaterial
+	pSpec := pro.Status.PipelineSpec
+	if pSpec != nil {
+		pipelineTasks := append(pSpec.Tasks, pSpec.Finally...)
+		for _, t := range pipelineTasks {
+			tr := pro.GetTaskRunFromTask(t.Name)
+			// Ignore Tasks that did not execute during the PipelineRun.
+			if tr == nil || tr.Status.CompletionTime == nil {
+				logger.Infof("taskrun status not found for task %s", t.Name)
+				continue
+			}
+
+			// add step images
+			if err := taskrun.AddStepImagesToMaterials(tr.Status.Steps, &mats); err != nil {
+				return mats, nil
+			}
+
+			// add sidecar images
+			if err := taskrun.AddSidecarImagesToMaterials(tr.Status.Sidecars, &mats); err != nil {
+				return mats, nil
+			}
+
+			// add remote task configsource information in materials
+			if tr.Status.Provenance != nil && tr.Status.Provenance.ConfigSource != nil {
```

Comment:
Should we also add to materials the Provenance info for the PipelineRun itself?

Reply:
you mean the config source info of the pipeline run?

Reply:
Right. My reasoning is that if the Pipeline definition comes from a Tekton Bundle, for example, then that image is yet another artifact that influenced the build. Wdyt?

Reply:
We can include it here. Though that information would be captured in configSource as well. If we are ok with that duplication then I'm happy to include it here.

Reply:
I think duplication is ok. In fact, the image references from the TaskRun status are already included in the provenance under buildConfig, as the discussion in https://github.com/tektoncd/chains/issues/642 shows. To be honest, I'm not sure what the right call is in this case.

Reply:
SGTM! I will include it here.

Reply:
Done!